.github/workflows/build-test-measure.yml

```diff
@@ -5,93 +5,427 @@ on:
+            coverage: true
+
+          - php: '7.0'
+            wp: '5.1'
+            coverage: false
+            install-pwa-plugin: true
+
+          - php: '5.6'
+            wp: '5.0'
+            coverage: false
+            install-pwa-plugin: true
+
+          - php: '5.6'
+            wp: '4.9'
+            external-http: true
+            install-pwa-plugin: false
+            coverage: false
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
         with:
-          name: amp-${{ steps.retrieve-branch-name.outputs.branch_name }}-${{ steps.retrieve-git-sha-8.outputs.sha8 }}-dev
-          path: builds/dev
+          php-version: ${{ matrix.php }}
+          extensions: curl, date, dom, gd, iconv, json, libxml, mysql, spl
+          coverage: ${{ matrix.coverage && 'pcov' || 'none' }}
+          ini-values: pcov.directory=.
+          tools: composer:v1, cs2pr
+
+      - name: Shutdown default MySQL service
+        run: sudo service mysql stop
+
+      - name: Verify MariaDB connection
+        run: |
+          while ! mysqladmin ping -h"127.0.0.1" -P"${{ job.services.mysql.ports[3306] }}" --silent; do
+            sleep 1
+          done
+
+      - name: Get Composer Cache Directory
+        id: composer-cache
+        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+      - name: Configure Composer cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-composer-
+
+      - name: Install Composer dependencies
+        run: composer install --prefer-dist --ignore-platform-reqs --no-suggest --no-progress --no-interaction
+
+      - name: Get npm cache directory
+        id: npm-cache
+        run: echo "::set-output name=dir::$(npm config get cache)"
+
+      - name: Configure npm cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.npm-cache.outputs.dir }}
+          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-npm-
+
+      - name: Install Node dependencies
+        run: npm ci
+        env:
+          CI: true
+
+      - name: Build plugin
+        run: |
+          npm run build:js
+          npm run build:css
+
+      # Scan the logs for failing tests and surface that information by creating annotations and log file decorations.
+      - name: Setup problem matcher to provide annotations for PHPUnit
+        # The JSON file is provided by the `shivammathur/setup-php` action. See https://github.com/shivammathur/setup-php#problem-matchers.
+        run: echo "::add-matcher::${{ runner.tool_cache }}/phpunit.json"
+
+      - name: Set up tests
+        # TODO: Add step to install plugins instead of relying on .dev-lib
```

Comment:
Correct me if I'm wrong, but couldn't we do this now by moving the code from `.dev-lib` into `install-wp-tests.sh`? It seems like this PR gets us 90% of the way toward removing the `wp-dev-lib` dependency. Can we get to 100%?

Reply:
Good question. The original intention was to remove the `wp-dev-lib` dependency but I realized it is being used for generating the readme (@westonruter any other uses I'm missing?) which also relies on the `.dev-lib` file, so I didn't spend much time looking into migrating the installation of the plugins just yet. I suppose that still can be done and then we could leave the `.dev-lib` file in place until all use cases for it are migrated.